examples/s_expressions.rs

```diff
@@ -16,18 +15,43 @@ use rowan::SmolStr;
 
 /// Let's start with defining all kinds of tokens and
 /// composite nodes.
-use rowan::SyntaxKind;
 
-const L_PAREN: SyntaxKind = SyntaxKind(0); // '('
-const R_PAREN: SyntaxKind = SyntaxKind(1); // ')'
-const WORD: SyntaxKind = SyntaxKind(2); // '+', '15'
-const WHITESPACE: SyntaxKind = SyntaxKind(3); // whitespaces is explicit
-const ERROR: SyntaxKind = SyntaxKind(4); // as well as errors
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+enum Lang {}
+impl rowan::Language for Lang {
+    fn debug_kind(kind: &SyntaxKind, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let tag = match *kind {
```

Comment:
I definitely *definitely* want a macro generating this. Not procedural ones or anything too fancy, just
```rust
rowan_enum! {
    onto Lang,
    ---
    L_PAREN = 0,
    R_PAREN, // implicit = 1 and so on
}
```
=>
```rust
const L_PAREN: SyntaxKind = SyntaxKind::new(0);
const R_PAREN: SyntaxKind = SyntaxKind::new(1);

impl rowan::Language for Lang {
    fn debug_kind(kind: &SyntaxKind, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "{}", match *kind {
            // actually converting an identifier to a string with a
            // macro is not well-defined, but for such very simple
            // cases as this one I think one can't possibly blame us
            // if we rely on `stringify!`
            L_PAREN => "L_PAREN",
            R_PAREN => "R_PAREN",
        })
    }
}
```

Reply:
sgtm!

Reply:
Heh, I had made Kind an associated type: in rust-analyer, kinds live in a crate which doesn't know about the sytanax tree at all.

Reply:
That works, then I could switch back to an enum I guess